src/scipp/core/__init__.py

```diff
@@ -30,6 +30,13 @@
 setattr(Dataset, 'rename', _rename_dataset)
 del _rename_dims, _rename_variable, _rename_data_array, _rename_dataset, cls
 
+for cls in (Variable, DataArray, Dataset):
+    docstr = """Total bytes consumed by the object. """ + \
+        f"""See :py:func:`scipp.{cls.__name__}.underlying_size` """ + \
+        """for more information."""
+    setattr(cls, 'nbytes', property(cls.underlying_size, doc=docstr))
+del cls, docstr
+
```

Comment:
I don't think I would add this, in particular not with a behavior that is different from Numpy's (which returns the bytes taken by the elements, not the whole array).